### DISCUSSION TITLE
Suggestions for low spec PC / GPUs ?

### DISCUSSION THREAD

**Discussion by HolgiB** [oobabooga/text-generation-webui#1704] [Q&A] [answered]
Hi there,

I am currently exploring AI based chatbots a bit and got somewhat lost in between all those language models, system requirements, 4 bit quantization stuff, etc.

I have three systems:
Dual core AMD / 8 GB RAM / GeForce 730GT
Quad Core AMD / 16 GB RAM / GeForce 1050 Ti with 4 GB VRAM
Six Core AMD / 32 GB RAM / GeForce 1060 Ti with 6 GB VRAM

All three systems run Linux in some flavor.

What are reasonable models to try out to get a reasonable results in terms of runtime and answer precsion ?

All those models referenced over at Redit in some discussion thread still leave me unsure what files to grab. Plus  when grabbing a model they often do not state wether they are GPU and / or CPU optimized. 

What I also do not get in the first place: When I launch the download-model.py to grab models from huggingface.co it gives me missing module errors. I guess this is because there is a Python virtual environment set up ? How do I start this from bash ?

Thanks in advance,
Holger

**Comment by Zhincore** [comment, marked as answer]
GPTQ 4bit quantized models should consume less resources, so those are something you might want to look for; with the parameter `--pre_layer` you may be able to fit some of them into the 6 GB VRAM. 
I really like Vicuna-13B-GPTQ, she and her context somewhat fit into 12 GB with `--pre_layer 32`. You may have some luck with Vicuna-7B-GPTQ, from what I understand she should be smaller.

Also my findings are that CPU models are marked with "GGML" and end with `.bin` or the README mentions that it's made for llama.cpp. GPU models seem to end with `.pt` or better `.safetensors`, not sure if it's a rule, though.

Python venv setup and activation (using conda in this case) is described in the [README which you can find here.](https://github.com/oobabooga/text-generation-webui#manual-installation-using-conda).

**Comment by HolgiB** [comment]
Hey there,

thanks for taking the time to answer. 

I found also this part of the guide reference in the oobabooga manual:
https://github.com/oobabooga/text-generation-webui/blob/main/docs/Low-VRAM-guide.md

I guess I will go for "python server.py --auto-devices --gpu-memory 3500MiB"

As for Miniconda and the virtual environment I was not really aware of what happens in the background. Espescially since I have some Python experiences but neither worked much with virtual environments, nor have I touched Conda / Miniconda in any way.

For the models I have also found out about the ggml.bin files for CPU versus the completely different looking files for GPU. Anyway, I would be happy if I can get Vicuna 7b GPU-accelerated at least on my two boxes with GeForce 10 GPUs. 

In the meantime I will try to figure out the differences between all those different models.

**Reply by tlholley74** [reply]
MLC LMM workd well on that card. It runs Vicuna 1.7b. Here's the link and full instructions. https://www.cloudbooklet.com/how-to-install-mlc-llm/

**Comment by xianmbga** [comment]
Hi im new in this. I want to have an LLM in my computer:

CPU	Intel(R) Core(TM) i7 CPU         870  @ 2.93GHz   2.93 GHz 64 bits
RAM 	10,0 GB
GPU nvidia GTX 1060 Gb VRAM

I ve installed TheBloke_Marcoroni-7B-v3-AWQ model, and when i load it, this happens:

torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 128.00 MiB. GPU 0 has a total capacty of 6.00 GiB of which 0 bytes is free. Of the allocated memory 9.26 GiB is allocated by PyTorch, and 325.41 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF


Because of this i installed mistral-7b-instruct-v0.1.Q5_0.gguf but i have some problems too:
2023-12-11 13:50:09 INFO:llama.cpp weights detected: models\mistral-7b-instruct-v0.1.Q5_0.gguf
2023-12-11 13:50:09 ERROR:Failed to load the model.
Traceback (most recent call last):
  File "E:\ChatGPTpirata\text-generation-webui\modules\ui_model_menu.py", line 209, in load_model_wrapper
    shared.model, shared.tokenizer = load_model(shared.model_name, loader)
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ChatGPTpirata\text-generation-webui\modules\models.py", line 88, in load_model
    output = load_func_map[loader](model_name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ChatGPTpirata\text-generation-webui\modules\models.py", line 253, in llamacpp_loader
    model, tokenizer = LlamaCppModel.from_pretrained(model_file)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\ChatGPTpirata\text-generation-webui\modules\llamacpp_model.py", line 54, in from_pretrained
    Llama = llama_cpp_lib().Llama
            ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'Llama'

I dont understand, can you help me please?
